juriscraper/scotus/scotus_email.py

```diff
@@ -0,0 +1,176 @@
+        `datetime`, returning `None` if the header is absent or parsing fails
+        and logging an appropriate error.
+
+        :return: `datetime` or `None` if unable to extract or parse the "Date" header.
+        """
+        if self.is_valid:
+            message_date = self.message.get("Date")
+
+            if message_date:
+                try:
+                    return datetime.strptime(
+                        message_date, "%a, %d %b %Y %H:%M:%S %z"
+                    )
+                except ValueError:
+                    logger.error(
+                        "Unable to parse 'Date' header of email (value is '%s')",
+                        message_date,
+                    )
+                    return None
+            else:
+                logger.error("Unable to find 'Date' header of email")
+                return None
+        else:
+            return None
+
+    def _parse_filing_name(self) -> Optional[str]:
+        """Extract the docket entry title from the email body using a regex
+        and clean it with the `clean_string` method. Return `None` and log
+        an error if the regex fails to match the email.
+
+        :return: Clean docket entry title or `None` if unable to extract the title.
+        """
+        if self.is_valid:
+            text = self.tree.text_content()
+            match = self.TITLE_REGEX.match(text)
+
+            if match is None:
+                logger.error(
+                    "Unable to find match for docket entry title regex in email"
+                )
+                return None
+            else:
+                return clean_string(match.group(1))
+        else:
+            return None
+
+    def _parse_case_name(self) -> Optional[str]:
+        """Extract the case name from the first `<a>` tag in the email body
+        and uses the `harmonize` method to clean it.
+
+        :return: Cleaned case name or `None` if unable to extract the case name.
+        """
+        if self.is_valid:
+            return harmonize(self.tree.findtext(".//a"))
+        else:
+            return None
+
+    def _parse_case_link(self) -> Optional[str]:
+        """Extract the `href` attribute from the first `<a>` tag in the email
+        body, which should be the link to the case.
+
+        :return: Link to the case or `None` if unable to extract the link.
+        """
+        if self.is_valid:
+            return self.tree.find(".//a").get("href")
+        else:
+            return None
+
+    def _parse_docket_number(self) -> Optional[str]:
+        """Extract the docket number using the `href` attribute from the first
+        `<a>` tag in the email body.
+
+        :return: Docket number or `None` if unable to extract the docket number.
+        """
+        if self.is_valid:
+            file = parse_qs(
+                urlparse(self.tree.find(".//a").get("href")).query
+            )["filename"][0]
+            (docket_number, _) = os.path.splitext(os.path.basename(file))
```

Comment:
We can simplify this line by using  `pathlib.Path`

```suggestion
            docket_number = Path(file).stem
```

In CL, we also prefer using `pathlib` instead of `os`.